x-pack/plugins/ml/public/alerting/anomaly_detection_alerts_table/render_cell_value.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash';
+import React, { type ReactNode } from 'react';
+import { isDefined } from '@kbn/ml-is-defined';
+import { ALERT_DURATION, ALERT_END, ALERT_START } from '@kbn/rule-data-utils';
+import type { GetRenderCellValue } from '@kbn/triggers-actions-ui-plugin/public';
+import { FIELD_FORMAT_IDS, FieldFormatsRegistry } from '@kbn/field-formats-plugin/common';
+import { getSeverityColor } from '@kbn/ml-anomaly-utils';
+import { EuiHealth } from '@elastic/eui';
+import {
+  alertFieldNameMap,
+  ALERT_ANOMALY_SCORE,
+  ALERT_ANOMALY_TIMESTAMP,
+} from '../../../common/constants/alerts';
+import { getFieldFormatterProvider } from '../../application/contexts/kibana/use_field_formatter';
+
+interface Props {
+  columnId: string;
+  data: any;
+}
+
+export const getMappedNonEcsValue = ({
+  data,
+  fieldName,
+}: {
+  data: any[];
+  fieldName: string;
+}): string[] | undefined => {
+  const item = data.find((d) => d.field === fieldName);
+  if (item != null && item.value != null) {
+    return item.value;
+  }
+  return undefined;
+};
+
+const getRenderValue = (mappedNonEcsValue: any) => {
+  // can be updated when working on https://github.com/elastic/kibana/issues/140819
+  const value = Array.isArray(mappedNonEcsValue) ? mappedNonEcsValue.join() : mappedNonEcsValue;
+
+  if (!isEmpty(value)) {
+    if (typeof value === 'object') {
+      return JSON.stringify(value);
+    }
+    return value;
+  }
+
+  return '—';
+};
+
+export const getRenderCellValue = (fieldFormats: FieldFormatsRegistry): GetRenderCellValue => {
+  const alertValueFormatter = getAlertFormatters(fieldFormats);
+
+  return ({ setFlyoutAlert }) =>
+    (props): ReactNode => {
+      const { columnId, data } = props as Props;
+      if (!isDefined(data)) return;
+
+      const mappedNonEcsValue = getMappedNonEcsValue({
+        data,
+        fieldName: columnId,
+      });
+      const value = getRenderValue(mappedNonEcsValue);
+
+      return alertValueFormatter(columnId, value);
+    };
+};
+
+export function getAlertFormatters(fieldFormats: FieldFormatsRegistry) {
```

Comment:
The date formatting in this case respects the `dateFormat` pattern defined in the Kibana Settings. I reckon it's way we should use field formatting defined by the user, because the preferences might be different per region. Also some users may prefer a better precision.